Added scripts for on-the-fly log file compression and reading.

dmitrijus opened the pull request:
Added scripts for on-the-fly log file compression (ztee.py) and reading (ztail.py).

DQMMonitoringService now outputs per-ls stats, improved event rate calculation.

Refactored esMonitoring.py to pass environment variable (including GZIP_LOG).
And added on-fly-compression option.

I had to remove -Werror=unused-variable, otherwise build fails for this package:
there is an unused variable in boost include.

Commit:
Added scripts for on-the-fly log file compression (ztee.py) and reading (ztail.py).

DQMMonitoringService now outputs per-ls stats, improved event rate calculation.

Refactored esMonitoring.py to pass environment variable (including GZIP_LOG).
Now with sockets!

I had to remove -Werror=unused-variable, otherwise build fails for this package:
there is an unused variable in boost include.

cmsbuild commented:
A new Pull Request was created by @dmitrijus (Dmitrijus) for CMSSW_7_4_X.

Added scripts for on-the-fly log file compression and reading.

It involves the following packages:

DQMServices/StreamerIO


@cmsbuild, @danduggan, @nclopezo, @deguio can you please review it and eventually sign? Thanks.
@barvic this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

deguio commented:
I know this one was tested on the DQM playback at P5.
was it submitted for 75 as well?

dmitrijus commented:
I will make port to 75 after we finish migration at P5.

degano commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 04397d983998e20a83b7b7c1ebc19ec352c34b5c
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9332/5204/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9332/5204/summary.html

Commit:
Fix a minor, but a critical mistake.

Mistake caused the esMonitoring.py to waste cpu cycles,
but not properly signaling write-non-readiness to select call.

deguio commented:
+1
please submit to 75x as well. the automatic fwd port has been disabled

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_4_X IBs unless changes or unless it breaks tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

cmsbuild commented:
+1
Tested at: c544edee5f42d60b2c66f1b28c2b8667b7d796fc
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9332/10023/summary.html

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_4_X IBs unless changes (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9332/10023/summary.html

There are some workflows for which there are errors in the baseline:
 50202.0 step 3
 The results for the comparisons for these workflows could be incomplete 
This means most likely that the IB is having errors in the relvals.The error does NOT come from this pull request

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_4_X IBs unless changes (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

davidlange6 commented on DQMServices/StreamerIO/plugins/BuildFile.xml:
```diff
@@ -8,4 +8,5 @@
 
 <library   file="*.cc" name="DQMServicesStreamerIOPlugins">
   <flags   EDM_PLUGIN="1"/>
+  <flags   REM_CXXFLAGS="-Werror=unused-variable"/>
```
@dmitrijus - why is this needed?

dmitrijus replied:
Hi David,

because otherwise the build fails in a boost include:
/opt/offline/slc6_amd64_gcc491/external/boost/1.51.0-cms2/include/boost/asio/error.hpp:244:45: error: 'boost::asio::error::system_category' defined but not used [-Werror=unused-variable]
 static const boost::system::error_category& system_category

Dmitrijus

davidlange6 commented:
+1
